docs/contracts.rst

```diff
@@ -16,51 +16,23 @@ inaccessible.
 Creating Contracts
 ******************
 
-Contracts can be created "from outside" or from Solidity contracts.
+Contracts can be created "from outside" via Ethereum transactions or from within Solidity contracts.
+
+IDEs, such as `Remix <https://remix.ethereum.org/>`_, make the creation process seamless using UI elements.
+
+Creating contracts programatically on Ethereum is best done via using the JavaScript API `web3.js <https://github.com/etherem/web3.js>`_.
+As of today it has a method called `web3.eth.Contract <https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html#new-contract>`_
+to facilitate contract creation.
```

Comment:
Perhaps we could provide a full js snippet that does this in the future and also invokes `solc-js`. Perhaps something like https://github.com/chriseth/scrypt-interactive/blob/master/test/index.js